examples/PyTorch/Inference/hf_transformers/blade_adapter.py

```diff
@@ -0,0 +1,305 @@
+# Copyright 2022 The BladeDISC Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from copy import deepcopy
+from inspect import signature
+from typing import Any, Dict, List, NamedTuple, Optional, Tuple, Type, Union
+
+import torch
+import torch_blade
+from transformers import (AutoConfig, AutoFeatureExtractor, AutoTokenizer,
+                          FeatureExtractionMixin, Pipeline, PretrainedConfig,
+                          PreTrainedModel, PreTrainedTokenizer, ProcessorMixin)
+from transformers import pipeline as hf_pipeline
+from transformers.feature_extraction_utils import PreTrainedFeatureExtractor
+from transformers.models.auto.auto_factory import _get_model_class
+from transformers.onnx.features import FeaturesManager
+from transformers.onnx.utils import get_preprocessor
+from transformers.pipelines import (check_task, get_default_model_and_revision,
+                                    get_task, infer_framework_load_model)
+from transformers.utils import TensorType
+from transformers.utils.generic import ModelOutput
+
+LOGGER = logging.getLogger(__name__)
+
+Preprocessor = Union[PreTrainedTokenizer,
+                     FeatureExtractionMixin, ProcessorMixin]
+
+
+class ModelInfo(NamedTuple):
+    task: str
+    id_or_path: str
+    config: PretrainedConfig
+    targeted_task: Dict[str, Any]
+    model_class: Type[PreTrainedModel]
+    dummy_inputs: Dict[str, torch.Tensor]
+    input_order: List[str]
+    default_kwargs: Dict[str, Any]
+    output_names: List[str]
+
+
+# pipeline task to model task mapping
+TASK_MAP = {
+    'audio-classification': 'sequence-classification',  # ?
+    'automatic-speech-recognition': 'speech2seq-lm',
+    'feature-extraction': 'default',  # ?
+    'text-classification': 'sequence-classification',
+    'token-classification': 'token-classification',
+    'question-answering': 'question-answering',
+    # 'table-question-answering': '',
```

Comment:
any problem with commented tasks?

Reply:
I've not verified these pipeline tasks to model tasks mapping.  I added a TODO to support these tasks in the updated commit.